serializer/src/commanddecoder.c

```diff
@@ -191,7 +191,7 @@ static EXECUTE_COMMAND_RESULT DecodeAndExecuteModelAction(COMMAND_DECODER_HANDLE
 #ifdef _MSC_VER
 #pragma warning(suppress: 6324) /* We intentionally use here strncpy */ 
 #endif
-        if (strncpy(tempStr, actionName, strLength - 1) == NULL)
+        if (memcpy(tempStr, actionName, strLength - 1) == NULL)
```

Comment:
Hi @pboettch ,
there was a bug in this part of the code that we had planned to fix, but with your change now it needs to be fixed in a different way.

To allow us to merge your change in and record your contribution, could you make the following changes please?

1. Change the memcpy call to: ```if (memcpy(tempStr, actionName, strLength + 1) == NULL)```

(note the strLength + 1 instead of strLength - 1)

and

2. Remove ```tempStr[strLength - 1] = 0;``` (on line 209 below).

Reply:
@pboettch I shall take your branch and apply the changes that Ewerton suggested.  I need to make the change as soon as possible.  I will merge this change into the SDK.

Thanks,
Jelani

Reply:
Just got up. I see you merged it into your memcpy-fix branch. Do not hesitate to request more changes. If I can be helpful, I will be.